app/services/publisher_balance_getter.rb

```diff
@@ -0,0 +1,78 @@
+# Used by the PublisherWalletGetter to retrieve balances
+class PublisherBalanceGetter < BaseApiClient
+  attr_reader :publisher
+
+  def initialize(publisher:)
+    @publisher = publisher
+  end
+
+  def perform
+    return [] if publisher.channels.verified.empty?
+    return perform_offline if Rails.application.secrets[:api_eyeshade_offline]
+
+    channel_balances_response = connection.get do |request|
+      request.headers["Authorization"] = api_authorization_header
+      request.url("/v1/balances?account=#{URI.escape(publisher.owner_identifier)}#{URI.escape(channels_query_string)}")
+    end
+
+    channel_balances = JSON.parse(channel_balances_response.body)
+
+    # Eyeshade returns an empty response for verified channels
+    # with no balance, so we must fill in 0.00 balances
+
+    # Find verified channel IDs with no balance
+    verified_channel_ids = []
+    publisher.channels.verified.each {|channel| verified_channel_ids.push(channel.details.channel_identifier)}
+
+    channel_balances.each do |channel_balance|
+      channel_identifier = channel_balance["account"]
+      verified_channel_ids.delete(channel_identifier) if verified_channel_ids.include?(channel_identifier)
+    end
+
+    # Set balances to 0.00
+    verified_channel_ids.each do |id|
+      channel_balance = {
+        "account" => "#{id}",
+        "balance" => "0.00"
+      }
+
+      channel_balances.push(channel_balance)
+    end
+
+    channel_balances
+  end
+
+  def perform_offline
+    channels_json = []
+    @publisher.channels.verified.each do |channel|
+      channels_json.push({
+        "account" => "#{channel.details.channel_identifier}",
+        "balance" => rand(0..3000)
+      })
+    end
+
+    channels_json
+  end
+
+  private
+
+  def channels_query_string
+    return "" if publisher.channels.verified.count == 0
+
```

Comment:
You can simplify this block by doing:

publisher.channels.verified.select { |channel| "&account=#{channel_details.channel_identifier} }.reduce(:+)

Reply:
Hm, I don't think so since `publisher.channels.verified.select { |channel| "&account=#{channel_details.channel_identifier} }` returns an array of channels

Reply:
Yeah, but channel isn't implicitly convertible to string.  The above gives this error:

```NoMethodError Exception: undefined method `+' for #<Channel:0x00007fac383456e8>```

We could do this instead
```publisher.channels.verified.map { |channel| "&account=#{channel.details.channel_identifier}" }.reduce(:+)```, though I'm not sure this is more readable than the original.